prml/generic-asset/src/lib.rs

```diff
@@ -498,6 +528,27 @@ decl_storage! {
 	}
 }
 
+fn migrate_locks<T: Config>() {
+	#[allow(dead_code)]
+	mod inner {
+		use super::Config;
+		use crate::types::BalanceLock;
+		pub struct Module<T>(sp_std::marker::PhantomData<T>);
+		frame_support::decl_storage! {
+			trait Store for Module<T: Config> as GenericAsset {
+				pub Locks get(fn locks):
+					map hasher(blake2_128_concat) T::AccountId => Vec<BalanceLock<T::Balance>>;
+			}
+		}
+	}
+
+	let staking_asset_id = <StakingAssetId<T>>::get();
+	let old_locks = <inner::Locks<T>>::drain().collect::<Vec<_>>();
+	old_locks.iter().for_each(|(account_id, locks)| {
```

Comment:
thinking this would take elements 1 by 1 vs. allocate a new vec

```suggestion
	<inner::Locks<T>>::drain().iter().for_each(|(account_id, locks)| {
```

Reply:
I realised Locks<T> and inner::Locks<T> even though they have different layouts and different names, they are internally based on the same storage which is viewed differently. It wasn't sound to insert to Locks while inner::Locks was being drained. It could actually make the integration test fail. So I had to drain the previous content of Locks in a separate step.